sonarcloud/create_sonarcloud_projects-token.sh

```diff
@@ -67,32 +92,40 @@ curl_post() {
 curl_get() {
   local api_path="$1"
   local api_opts="$2"
+  local dry="${3:-${DRY_RUN}}"
 
-  if ! ${DRY_RUN}; then
+  if ! ${dry}; then
     curl -sSL \
       --request GET \
       --header "Content-Type: application/x-www-form-urlencoded" \
       -u "${SONAR_TOKEN}": \
      "${SONAR_API_BASE_URL}/${api_path}?${api_opts}"
   else
-    echo "DRY-RUN: curl -sSL --request POST --header \"Content-Type: application/x-www-form-urlencoded\" -u \"${SONAR_TOKEN}\": -d \"${data}\" \"${SONAR_API_BASE_URL}/${api_path}\"" >&2
+    echo "DRY-RUN: curl -sSL --request POST --header \"Content-Type: application/x-www-form-urlencoded\" -u \"${SONAR_TOKEN}\": \"${SONAR_API_BASE_URL}/${api_path}?${api_opts}\"" >&2
     echo "{}"
   fi
 }
 
 get_projects() {
   local sonar_organization="$1"
-  curl_get  "projects/search" "organization=${sonar_organization}&ps=500"| jq -r '.components[].key'
+  curl_get  "projects/search" "organization=${sonar_organization}&ps=500" false | jq -r '.components[].key'
 }
 
 create_token() {
   local token_name="Analyze \"${1:-}\""
   local suffix="${2:-}"
-
   echo "Creating SonarCloud token:"
   reply="$(curl_post "name=${token_name}" 'user_tokens/generate')"
   if echo "${reply}" | jq -e 'has("errors")' > /dev/null ; then
-    echo "${reply}" | jq -r '.errors[].msg'
+    error_msg=$(echo "${reply}" | jq -r '.errors[].msg')
+    echo "WARNING creating token: ${error_msg}"
+
+    generate_new_token=$(_question_true_false "revoke token and generate new one")
+    if [ "$generate_new_token" = true ]
+    then
```

Comment:
Can be simplified to:
```
    if $(_question_true_false "revoke token and generate new one"); then
```

Reply:
I changed to 

> if question_true_false "revoke token and generate new one"; then

Otherwise shellcheck raises: [SC2091](https://www.shellcheck.net/wiki/SC2091)

Reply:
PR: https://github.com/eclipse-cbi/ci-admin/pull/64